SensusUI/ProtocolsPage.cs

```diff
@@ -152,35 +156,59 @@ public ProtocolsPage()
                         actions.Add("Ungroup");
                 }
 
+                bool isPriorToStart = !(DateTime.Now > new DateTime(selectedProtocol.StartDate.Year, selectedProtocol.StartDate.Month, selectedProtocol.StartDate.Day, selectedProtocol.StartTime.Hours, selectedProtocol.StartTime.Minutes, 0));
+                if (!selectedProtocol.ScheduledToStart && !selectedProtocol.Running && !selectedProtocol.StartImmediately && isPriorToStart)
+                {
+                    actions.Remove("Start");
+                    actions.Insert(0, "Schedule To Start");
+                }
+
                 if (selectedProtocol.Running)
+                    actions.Remove("Schedule To Start");
                     actions.Add("Status");
 
+                if (!selectedProtocol.Running && selectedProtocol.ScheduledToStart)
+                {
+                    actions.Remove("Start");
+                    actions.Remove("Schedule To Start");
+                    actions.Insert(0, "Cancel Scheduled Start");
+                }
+
                 actions.Add("Delete");
 
                 string selectedAction = await DisplayActionSheet(selectedProtocol.Name, "Cancel", null, actions.ToArray());
 
-                // must reset the protocol selection manually
-                SensusServiceHelper.Get().RunOnMainThread(() =>
+                // must reset the protocol select manually
+                Device.BeginInvokeOnMainThread(() =>
                 {
                     _protocolsList.SelectedItem = null;
                 });
 
-                if (selectedAction == "Start")
+                if (selectedAction == "Start" || selectedAction == "Schedule To Start")
                 {
                     selectedProtocol.StartWithUserAgreementAsync(null, () =>
                     {
                         // rebind to pick up color and running status changes
-                        SensusServiceHelper.Get().RunOnMainThread(Bind);
+                        Refresh();
                     });
                 }
+                else if (selectedAction == "Cancel Scheduled Start")
+                {
+                    if (await DisplayAlert("Confirm Cancel", "Are you sure you want to cancel " + selectedProtocol.Name + "?", "Yes", "No"))
+                    {
+                        selectedProtocol.CancelScheduledStart();
+                        // rebind to pick up color and running status changes
+                        Refresh();
+                    }
+                }
```

Comment:
Won't `Protocol.Stop` cancel any scheduled starts, obviating the need for this action? The Stop action can cover both cases in the same way that Start can.